genisys/modules/firstboot/hello.py

```diff
@@ -0,0 +1,49 @@
+import json
+from pathlib import Path
+from typing_extensions import Self
+
+from genisys.modules.base import Module
+from genisys.config_parser import YAMLParser
+
+class Hello(Module):
+    """
+    The FirstBoot class generates a shell script for the 
+    client machines which sends a POST request to the 
+    Genisys server containing the IP address and hostname 
+    of the client machine for use in the Ansible inventory file. 
+    """
+
+    def __init__(self: Self, config: YAMLParser):
+        self.config = {}
+        self.config["scripts"] = config.get_section("Scripts")
+        self.config["network"] = config.get_section("Network")
+    #end __init__
+
+    def generate(self: Self):
+        content = []
+
+        json_body = {
+            'message' : 'hello', 
+            'ip' : '$(hostname -I)',
+            'hostname' : '$(hostname)'
+        }
+
+        # Not sure of json.dumps() is necessary, will need testing
+        json_object = json.dumps(json_body)
+
+        content.append("#!/bin/bash")
+
+        # Command to send POST request to Genisys server (Subject to change)
+        curl_command = "curl -X POST -H \"Content-Type: application/json\" -d" + "\"" + json_object + "\"" + self.config["network"]["ip"]
```

Comment:
Just from looking at it, this format looks a little weird. Filling in some placeholders, it looks like it'll produce `curl -X POST -H "Content-Type: application/json" -d"{"message": "hello"}"10.0.01`. The big problem is that last block, The double quotes that `json.dumps` puts around the json strings are going to unquote the json value so bash is gonna read this as `{message: hello}10.0.0.1`. The IP is easy, just add another space. The quotes are another thing.

It might work if you can find a way to have `json.dumps` output single quotes, but without testing it myself, that could affect the string interpolation you're using to get the IP and hostname. It might make more sense to use a tool like [`jq`](https://jqlang.github.io/jq/) to write a temporary JSON file then reference it in the `curl` command with `@filename`. Relevant from `man curl`: `If you start the data with the letter @, the rest should be a file name to read the data from, or a single dash (-) if you want curl to read the data from stdin.`

Quoting is a whole thing in Bash. Checkout the [GNU Bash Manual](https://www.gnu.org/software/bash/manual/html_node/Quoting.html) for all the rules.

Reply:
Fixed formatting, now running the outputted script correctly runs bash commands